### DISCUSSION TITLE
how to sync a localStorage variable with x-show

### DISCUSSION THREAD

**Discussion by playgroundctl** [alpinejs/alpine#1599] [1. Help] [answered]
Hi,

Given some html
```
<div x-data="{ running: localStorage.getItem('running') }" x-show="running">
				      <span class="relative inline-flex rounded-md shadow-sm w-full">
					      <p class="inline-flex justify-center items-center px-4 py-2 border border-transparent shadow-sm text-xs font-medium rounded-md text-white bg-gray-500 dark:bg-gray-200 dark:text-gray-500 w-full"> 
								  <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6 text-white dark:text-gray-500" fill="none" viewBox="0 0 24 24" stroke="currentColor">
								    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
								  </svg>
									&nbsp;&nbsp;Running...
							  </p>
					      <span class="flex absolute h-3 w-3 top-0 right-0 -mt-1 -mr-1">
					        <span class="animate-ping absolute inline-flex h-full w-full rounded-full bg-yellow-400 opacity-75"></span>
					        <span class="relative inline-flex rounded-full h-3 w-3 bg-yellow-400"></span>
					      </span>
					    </span>
				  	</div>
```

I wish to toggle the visibility `x-show="running"` when  `localStorage.getItem('running')` changes its value.
What triggers the `localStorage.getItem('running')` to change, is some external js.

I tried to use the `$watch` method directly with `x-data`, but it didn't work.

Could you please advise how to sync the visibility of `x-data="{ running: ... }"` with `localStorage.getItem('running')`?
(I mean the opposite of this blog post https://codewithhugo.com/alpinejs-localstorage-sessionstorage/)

Or, alternatively, using the `Alpine.store` how can one watch for a variable `$store.running.progress` and display (or not) the `x-data="{ running: ... }"` ?
```
  <script>
    document.addEventListener('alpine:init', () => {
        Alpine.store('running', {
            progress: false,
            state(value) {
                this.progress = value
            }
        })
    })
</script>
```

**Comment by danddanddand** [comment, marked as answer]
This isn't how I'd code it but should give you working reference. As far as I know you'd have to poll it if you’re trying to “watch” value from localStorage.

https://codepen.io/danddanddand/pen/BaWgyGw

**Reply by playgroundctl** [reply]
thank you, that was eye opening!

I understand that you would make it work with events, this is how I ended up doing it :)

```
window.dispatchEvent(new CustomEvent('running', { detail: true }));
```

```
<div x-data="{ progress: false }" x-on:running.window="progress = $event.detail" x-show="progress">
```
